File: flutter/lib/src/log4agent_config.dart
import 'log4agent_endpoint.dart';
import 'log4agent_redactor.dart';

class Log4AgentConfig {
  const Log4AgentConfig({
    this.enabled = true,
    this.endpoint = Log4AgentEndpoint.defaultAndroidEmulator,
    this.app = 'log4agent',
    this.deviceId = 'flutter',
    this.redactionEnabled = true,
    this.redactionKeys = Log4AgentRedactor.defaultSensitiveKeys,
  });

  final bool enabled;
  final String endpoint;
  final String app;
  final String deviceId;
  final bool redactionEnabled;
  final Set<String> redactionKeys;

  factory Log4AgentConfig.endpoint(
    String url, {
    bool enabled = true,
    bool redactionEnabled = true,
  }) {
    return Log4AgentConfig(
      enabled: enabled,
      endpoint: url,
      redactionEnabled: redactionEnabled,
    );
  }

  factory Log4AgentConfig.host(
    String host, {
    int port = 3100,
    String path = '/logs',
    String scheme = 'http',
    bool enabled = true,
    bool redactionEnabled = true,
  }) {
    return Log4AgentConfig(
      enabled: enabled,
      endpoint: Log4AgentEndpoint.fromHost(host, port: port, path: path, scheme: scheme),
      redactionEnabled: redactionEnabled,
    );
  }
}

File: ios/Sources/Log4Agent/Log4AgentConfig.swift
import Foundation

public struct Log4AgentConfig {
    public var enabled: Bool
    public var endpoint: URL
    public var app: String
    public var deviceId: String
    public var redactionEnabled: Bool
    public var redactionKeys: Set<String>

    public init(
        enabled: Bool = true,
        endpoint: URL = Log4AgentEndpoint.defaultURL,
        app: String = "log4agent",
        deviceId: String = "ios",
        redactionEnabled: Bool = true,
        redactionKeys: Set<String> = Log4AgentRedactor.defaultSensitiveKeys
    ) {
        self.enabled = enabled
        self.endpoint = endpoint
        self.app = app
        self.deviceId = deviceId
        self.redactionEnabled = redactionEnabled
        self.redactionKeys = redactionKeys
    }

    public static func host(
        _ host: String,
        port: Int = 3100,
        path: String = "/logs",
        scheme: String = "http"
    ) -> Log4AgentConfig {
        Log4AgentConfig(endpoint: Log4AgentEndpoint.fromHost(host, port: port, path: path, scheme: scheme))
    }
}
